src/ethereum/homestead/spec.py

```diff
@@ -649,9 +656,8 @@ def recover_sender(tx: Transaction) -> Address:
 
     ensure(v == 27 or v == 28)
     ensure(0 < r and r < SECP256K1N)
-
-    # TODO: this causes error starting in block 46169 (or 46170?)
-    # assert 0<s_int and s_int<(SECP256K1N//2+1)
+    # FIXME
+    # ensure(s == 0 or s > SECP256K1N // 2 -1)
```

Comment:
Not sure what's going wrong here. This causes tests to fail. I can address this in a separate PR.